src/main/java/seedu/address/model/person/Gender.java

```diff
@@ -9,13 +9,13 @@
  */
 public class Gender {
 
-    public static final String MESSAGE_CONSTRAINTS = "Gender can take any values, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Gender can only be either M or F";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final String VALIDATION_REGEX = "^M$|^F$";
```

Comment:
Should we allow users to input lowercases of  "M" and "F"? This gives users more flexibility. We can discuss this further.

Reply:
That's a good suggestion. We can implement this as an enhancement in future PRs